src/nni_manager/core/nniDataStore.ts

```diff
@@ -218,20 +218,20 @@ class NNIDataStore implements DataStore {
         if (trialJobEvents === undefined) {
             return result;
         }
-        const map: Map<string, TrialJobInfo> = this.getTrialJobsByReplayEvents(trialJobEvents);
+        const trialMap: Map<string, TrialJobInfo> = this.getTrialsByReplayEvents(trialJobEvents);
 
         const finalMetricsMap: Map<string, MetricDataRecord[]> = await this.getFinalMetricData(trialJobId);
 
-        for (const key of map.keys()) {
-            const jobInfo: TrialJobInfo | undefined = map.get(key);
-            if (jobInfo === undefined) {
+        for (const key of trialMap.keys()) {
+            const trialInfo: TrialJobInfo | undefined = trialMap.get(key);
+            if (trialInfo === undefined) {
                 continue;
             }
-            if (!(status !== undefined && jobInfo.status !== status)) {
-                if (jobInfo.status === 'SUCCEEDED') {
-                    jobInfo.finalMetricData = finalMetricsMap.get(jobInfo.id);
+            if (!(status !== undefined && trialInfo.status !== status)) {
+                if (trialInfo.status === 'SUCCEEDED') {
+                    trialInfo.finalMetricData = finalMetricsMap.get(trialInfo.id);
```

Comment:
what is the content of `finalMetricData` when there are multiple trials in a job?

Reply:
It's all metric data of a trial. No sure why it's called finalMetricData. I just change the variable name to differentiate it from jobMap.

Reply:
> what is the content of `finalMetricData` when there are multiple trials in a job?

It is a list of multiple final metrics for each trial

Reply:
I just checked, there are multiple final metrics from mutiple trials. so, `jobInfo` may be a better name.

Reply:
After this change, it only get one trial's final metric, not all.